src/hooks/useSignWithExtensionWallet.ts

```diff
@@ -0,0 +1,129 @@
+  const { walletKitPubKey, updateWalletKitPubKey } = account;
+  const networkPassphrase = getWalletKitNetwork(network.id);
+
+  const walletKitInstance = useContext(WalletKitContext);
+
+  const [signedTxXdr, setSignedTxXdr] = useState("");
+  const [successMsg, setSuccessMsg] = useState("");
+  const [errorMsg, setErrorMsg] = useState("");
+
+  const [isInProgress, setIsInProgress] = useState(false);
+
+  const SUCCESS_MSG = "1 signature added";
+
+  const reset = () => {
+    setSignedTxXdr("");
+    setSuccessMsg("");
+    setErrorMsg("");
+  };
+
+  const getErrorMsg = (error: any) =>
+    error?.message || error || "Something went wrong, please try again";
+
+  const signTx = useCallback(async () => {
+    if (isInProgress || !walletKitInstance?.walletKit) {
+      return;
+    }
+
+    setIsInProgress(true);
+
+    if (walletKitPubKey) {
+      try {
+        const result = await walletKitInstance.walletKit.signTransaction(
+          txXdr,
+          {
+            address: walletKitPubKey,
+            networkPassphrase,
+          },
+        );
+        setSignedTxXdr(result.signedTxXdr);
+        setSuccessMsg(SUCCESS_MSG);
+      } catch (error: any) {
+        if (error?.message) {
+          setErrorMsg(getErrorMsg(error));
+        }
+      }
+    } else {
+      // if a user didn't log in via stellar wallet kit in the main nav
+      // open a wallet kit modal to sign in
+      await walletKitInstance.walletKit.openModal({
+        onWalletSelected: async (option: ISupportedWallet) => {
+          try {
+            walletKitInstance.walletKit?.setWallet(option.id);
+            const addressResult =
+              await walletKitInstance.walletKit?.getAddress();
+
+            if (addressResult?.address) {
+              updateWalletKitPubKey(addressResult.address);
+
+              const result = await walletKitInstance.walletKit?.signTransaction(
+                txXdr,
+                {
+                  // You could send multiple public keys in case the wallet needs to handle multi signatures
+                  address: addressResult.address,
+                  networkPassphrase,
+                },
+              );
+
+              if (result?.signedTxXdr) {
+                setSignedTxXdr(result.signedTxXdr);
+                setSuccessMsg(SUCCESS_MSG);
+              } else {
+                throw {
+                  message: "Couldn’t sign with wallet, please try again",
+                };
+              }
+            }
+          } catch (error: any) {
+            if (error?.message) {
+              setErrorMsg(getErrorMsg(error));
+            }
+          }
+        },
+      });
+    }
+  }, [
+    isInProgress,
+    networkPassphrase,
+    txXdr,
+    updateWalletKitPubKey,
+    walletKitInstance.walletKit,
+    walletKitPubKey,
+  ]);
+
+  useEffect(() => {
+    if (isEnabled) {
+      reset();
+      signTx();
+    }
+  }, [isEnabled, signTx]);
+
+  useEffect(() => {
+    reset();
```

Comment:
is `isClear` supposed to be inside of this `useEffect`? I don't see it being used